test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/PageActionInvokerProviderTest.cs

```diff
@@ -334,6 +537,50 @@ public void OnProvidersExecuting_UpdatesEntriesWhenActionDescriptorProviderColle
                 NullLoggerFactory.Instance);
         }
 
+        private class TestSetPageWithModel
+        {
+            public TestSetPageModel Model { get; set; }
+        }
+
+        private class InheritsMethods : TestSetPageModel
+        {
+            public new void OnGet()
+            {
+
+            }
+        }
+
+        private class ProtectedPage
+        {
+            public ProtectedModel Model { get; set; }
+        }
+
+        private class ProtectedModel
+        {
+            protected void OnGet()
+            {
+
+            }
+
+            private void OnPost()
```

Comment:
Could you also add a `OnGet<TVal>` as a scenario. We should filter those out of discovery.